EIPS/eip-5792.md

```diff
@@ -236,23 +192,75 @@ This method accepts a call bundle identifier returned by a `wallet_sendCalls` ca
+
+#### `wallet_getCapabilities` RPC Specification
+
+Capabilities are returned in key/value pairs, with the key naming a capability and a value conforming to a shape defined for that name, in an object keyed to the relevant [EIP-155](./eip-155.md) `chainId` expressed in hexidecimal notation.
+Capabilities are nested in per-chain objects because wallets may support different capabilities across multiple chains authorized in a given session.
+
+```typescript
+type GetCapabilitiesParams = [`0x${string}`]; // Wallet address
+
+type GetCapabilitiesResult = Record<`0x${string}`, <Record<string, any>>; // Hex chain id
+```
+
+##### `wallet_getCapabilities` Example Parameters
+
+```json
+["0xd46e8dd67c5d32be8058bb8eb970870f07244567"]
+```
+
+##### `wallet_getCapabilities` Example Return Value
+
+The capabilities below are for illustrative purposes.
+
+```json
+{
+  "0x2105": {
+    "paymasterService": {
+      "supported": true
+    },
+    "sessionKeys": {
+      "supported": true
+    }
+  },
+  "0x14A34": {
+    "paymasterService": {
+      "supported": true
+    }
+  }
+}
+```
+
 ### `atomicBatch` Capability
 
-Indicates that a wallet can execute multiple calls atomically as part of a single transaction.
+Like the illustrative examples given above and other capabilities to be defined in future EIPs, the capability to execute calls delivered via the above-defined methods in a single transaction can be attested by the wallet in boolean form.
 
-If a wallet indicates it supports the `atomicBatch` capability, it MUST submit calls submitted with `wallet_sendCalls` as part of a single transaction.
+If a wallet has affirmatively expressed this `atomicBatch` capability to a calling application, it MUST submit calls submitted with `wallet_sendCalls` as part of a single transaction.
 
-#### `atomicBatch` Capaiblity Specification
+#### `atomicBatch` Capability Specification
 
 ```typescript
 type AtomicBatchCapability = {
   supported: true;
 };
 ```
 
-##### `wallet_getCapabilities` Example Return Value
+The only valid JSON-RPC values for this capability are `true` or `false`; if a returned value for "supported" value is typed as a string or number, it SHOULD be treated as malformed.
 
-For each chain on which a wallet can submit multiple calls atomically, the wallet SHOULD include an `atomicBatch` capability with a `supported` field equal to `true`.
+For each chain on which a wallet can submit multiple calls atomically, the wallet SHOULD include an `atomicBatch` capability with a `supported` field equal to `true`. For each chain where it cannot, it SHOULD return the same field as `false`. Unspecified support for a given, named chain SHOULD be interpreted as not exposing this information to the application, and as such, applications SHOULD NOT interpret an absent value as equivalent to `false`.
```

Comment:
i disagree with this. can we change to `...For each chain where it cannot, it MAY return the same field as 'false'. Unspecified support for a given, named chain SHOULD be interpreted as equivalent to 'false'.`

i dont think the wallet should have to be so explicit about this

Reply:
Good point, I think this was worded badly.  Please rereview after latest commit.

(Sidenote, if we are really going full OCap, would `authorized` make more sense than `supported` as the name for that boolean?)